tlspuffin/harness/openssl/src/put.c

```diff
@@ -308,9 +313,14 @@ AGENT openssl_create_server(const TLS_AGENT_DESCRIPTOR *descriptor)
     SSL_CTX_set_tmp_rsa(ssl_ctx, rsa);
 #endif
 
-    // Allow EXPORT in server
-    SSL_CTX_set_cipher_list(ssl_ctx, descriptor->cipher_string);
-    SSL_CTX_set_ciphersuites(ssl_ctx, descriptor->cipher_string);
+    if (descriptor->tls_version == V1_3)
+    {
+        SSL_CTX_set_ciphersuites(ssl_ctx, descriptor->cipher_string_tls13);
```

Comment:
Don't you want to first check that the `allow_setting_tls12/3_ciphers` capability is available?

Reply:
Capabilities are not accessible in this function, I don't really want to overhaul the whole c put architecture for this check. With the default TLS string the PUTs work normally but if somebody tries to set a different cipher on a PUT that don't support it, the PUT will already throw an error.